source-monday/source_monday/models.py

```diff
@@ -118,19 +119,120 @@ def get_errors(self) -> list[GraphQLError]:
+    """
+    List of activity log events that can be used to filter or process activity logs.
+
+    None of these are documented by Monday.com, but they are do exist from the
+    activity logs returned by the API.
+    """
+
+    ADD_OWNER = "add_owner"
+    ARCHIVE_GROUP_PULSE = "archive_group_pulse"
+    ARCHIVE_GROUP = "archive_group"
+    ARCHIVE_PULSE = "archive_pulse"
+    BATCH_CHANGE_PULSES_COLUMN_VALUE = "batch_change_pulses_column_value"
+    BOARD_DELETED = "board_deleted"
+    BATCH_CREATE_PULSES = "batch_create_pulses"
+    BATCH_DELETE_PULSES = "batch_delete_pulses"
+    BATCH_DUPLICATE_PULSES = "batch_duplicate_pulses"
+    BATCH_MOVE_PULSES_FROM_BOARD = "batch_move_pulses_from_board"
+    BATCH_MOVE_PULSES_FROM_GROUP = "batch_move_pulses_from_group"
+    BATCH_MOVE_PULSES_INTO_BOARD = "batch_move_pulses_into_board"
+    BATCH_MOVE_PULSES_INTO_GROUP = "batch_move_pulses_into_group"
+    BOARD_STATE_CHANGED = "board_state_changed"
+    BOARD_VIEW_ADDED = "board_view_added"
+    BOARD_VIEW_CHANGED = "board_view_changed"
+    BOARD_VIEW_ENABLED = "board_view_enabled"
+    BOARD_WORKSPACE_ID_CHANGED = "board_workspace_id_changed"
+    CHANGE_COLUMN_SETTINGS = "change_column_settings"
+    CREATE_COLUMN = "create_column"
+    CREATE_GROUP = "create_group"
+    CREATE_PULSE = "create_pulse"
+    DELETE_COLUMN = "delete_column"
+    DELETE_GROUP = "delete_group"
+    DELETE_PULSE = "delete_pulse"
+    MOVE_PULSE_FROM_BOARD = "move_pulse_from_board"
+    MOVE_PULSE_FROM_GROUP = "move_pulse_from_group"
+    MOVE_PULSE_INTO_BOARD = "move_pulse_into_board"
+    MOVE_PULSE_INTO_GROUP = "move_pulse_into_group"
+    MOVE_SUBITEM = "move_subitem"
+    REMOVE_OWNER = "remove_owner"
+    SUBSCRIBE = "subscribe"
+    UPDATE_BOARD_NAME = "update_board_name"
+    UPDATE_BOARD_NICKNAME = "update_board_nickname"
+    UPDATE_COLUMN_NAME = "update_column_name"
+    UPDATE_COLUMN_VALUE = "update_column_value"
+    UPDATE_GROUP_NAME = "update_group_name"
+    UPDATE_NAME = "update_name"
+    UNSUBSCRIBE = "unsubscribe"
+
+
+class ActivityLogEventType(StrEnum):
+    BOARD_CHANGED = auto()
+    BOARD_DELETED = auto()
+    ITEM_CHANGED = auto()
+    ITEM_DELETED = auto()
+    FULL_BOARD_ITEM_REFRESH = auto()
+
+
 class ActivityLogEventData(BaseModel, extra="allow"):
     board_id: int | None = None
     item_id: int | None = None
-    item_type: str | None = None
     pulse_id: int | None = None
     pulse_ids: list[int] | None = None
+    duplicated_pulse_ids: list[int] | None = None
+    subitem: int | None = None
 
 
 class ActivityLog(BaseModel, extra="allow"):
     entity: Literal["board", "pulse"]
-    event: str
+    event: ActivityLogEvents
     data: ActivityLogEventData
     created_at: str
+    # Set by the connector for logging purposes. These are only None
+    # when there are user unauthorized errors causing the response to be null.
+    query: str | None = None
+    query_variables: dict[str, Any] | None = None
```

Comment:
These can be `None` only if the full board in the response is `null`. See here: https://github.com/estuary/connectors/pull/3132/files#diff-f98022ade292bbaf8bdaf81f9ef171dc7c3d9c7a4dbd436d725ff8ed045baf2dR126-R127